NuGet/NuGetFeedClass.ps1

```diff
@@ -83,7 +85,20 @@ class NuGetFeed {
     }
 
     [hashtable[]] Search([string] $packageName) {
-        if ($this.searchQueryServiceUrl -match '^https://nuget.pkg.github.com/(.*)/query$') {
+        $useCache = $this.searchResultsCache.ContainsKey($packageName)
+        if ($useCache) {
+            # Clear cache older than 10 minutes
+            if ($this.searchResultsCache[$packageName].timestamp -lt (Get-Date).AddMinutes(-10)) {

```

Comment:
Please add the cache retention period to the bcContainerHelper settings in bc.helperfunctions.ps1 - this allows people to control this (or remove it) if needed - and make sure that 0 means do not use cache.